craft_store/http_client.py

```diff
@@ -0,0 +1,171 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Craft Store HTTPClient."""
+
+import logging
+import os
+from typing import Dict, Optional, Union
+
+import requests
+from requests.adapters import HTTPAdapter, Retry
+
+from . import errors
+
+logger = logging.getLogger(__name__)
+
+
+REQUEST_TOTAL_RETRIES = 8
+"""Amount of retries for a request."""
+REQUEST_BACKOFF = 0.2
+"""Backoff before retrying a request."""
+
+
+def _get_retry_value(
+    environment_var: str, default_value: Union[int, float]
+) -> Union[int, float]:
+    """Return the backoff to use in HTTPClient."""
+    environment_value = os.getenv(environment_var)
+    if environment_value is None:
+        return default_value
+
+    try:
+        value = int(environment_value)
+    except ValueError:
+        logger.debug(
+            "%r set to invalid value %r, setting to %r.",
+            environment_var,
+            environment_value,
+            default_value,
+        )
+        return default_value
+
+    if value <= 0:
```

Comment:
Shouldn't `REQUEST_TOTAL_RETRIES=0` be supported, to avoid retrying?

Reply:
I guess the same would apply for backoff

Reply:
this should now be allowed in https://github.com/canonical/craft-store/pull/1/commits/47bfbe2765dbf54339a035e1c0f5f5a8f07244b0